packages/core/src/providers/window/provider.tsx

```diff
@@ -0,0 +1,26 @@
+import { ReactNode, useState } from 'react'
+import { WindowContext } from './context'
+
+interface Props {
+  children: ReactNode
+}
+
+/**
+ * @internal Intended for internal use - use it on your own risk
+ */
+export function WindowProvider({ children }: Props) {
+  const [isActiveWindow, setActiveWindow] = useState(true)
+
+  document.addEventListener('visibilitychange', () => {
```

Comment:
Shouldn't this be in a useEffect?
We have no guarantee that this will be executed only once.

Reply:
It should be, thanks for catch:)